### PR TITLE
Refactor reference form (pagination)

### PR DESCRIPTION
**Pull request by jonkerz**
Refactor and refactor-ish to make all `Reference`s use pagination in a consistent manner.

Changes:

- Use `pagination` for all types of references. We don't need to separate book/article pagination in the form, and we want use pagination for `UnknownReferences` too. That only leaves `NestedReference`s which are currently using `references.pages_in` but never `references.pagination`, so we might as well merge the columns.
- Other smaller tweaks.

I believe the article/book pagination was a special case due to how fields are conditionally hidden/shown in the reference form, but it makes the class responsible for saving references more complicated than what it needs to be.

More work is required here, but I think this is enough for this session.

Short version: Use the `pagination` column and pagination form field for all references, including nested references (which previously used the `pages_in` column and always kept the `pagination` column blank).

### CODE REVIEW

**Comment by jonkerz** [comment]
Temporarily put on hold in order to deploy other big updates like gem upgrades.

**Comment by jonkerz** [comment]
```
NestedReference.where(pagination: [nil, '']).count
 => 1049 

NestedReference.where.not(pagination: [nil, '']).count
 => 0 
```
